tasks/Functions-admin/deployFullOracle.js

```diff
@@ -1,126 +1,124 @@
-      config.stalenessSeconds,
-      config.gasAfterPaymentCalculation,
-      config.weiPerUnitLink,
-      config.gasOverhead,
-      config.requestTimeoutSeconds
-    )
-    console.log('Registry configuration set')
-  
-    console.log('Deploying Functions oracle factory')
-    const oracleFactoryFactory = await ethers.getContractFactory('FunctionsOracleFactory')
-    const oracleFactory = overrides
-      ? await oracleFactoryFactory.deploy(overrides)
-      : await oracleFactoryFactory.deploy()
-    console.log(`Waiting for transaction ${oracleFactory.deployTransaction.hash} to be confirmed...`)
-    await oracleFactory.deployTransaction.wait(2)
-    console.log(`FunctionsOracleFactory deployed to ${oracleFactory.address} on ${network.name}`)
-  
-    console.log('Deploying Functions oracle')
-    const accounts = await ethers.getSigners()
-    const deployer = accounts[0]
-    const OracleDeploymentTransaction = overrides
-      ? await oracleFactory.deployNewOracle(overrides)
-      : await oracleFactory.deployNewOracle()
-    console.log(`Waiting for transaction ${OracleDeploymentTransaction.hash} to be confirmed...`)
-    const OracleDeploymentReceipt = await OracleDeploymentTransaction.wait(1)
-    const FunctionsOracleAddress = OracleDeploymentReceipt.events[1].args.oracle
-    const oracle = await ethers.getContractAt('FunctionsOracle', FunctionsOracleAddress, deployer)
-    console.log(`FunctionsOracle deployed to ${oracle.address} on ${network.name}`)
-  
-    // Set up Functions Oracle
-    console.log(`Accepting oracle contract ownership`)
-    const acceptTx = overrides
-      ? await oracle.acceptOwnership(overrides)
-      : await oracle.acceptOwnership()
-    console.log(`Waiting for transaction ${acceptTx.hash} to be confirmed...`)
-    await acceptTx.wait(1)
-    console.log('Oracle ownership accepted')
-  
-    console.log(`Setting DON public key to ${networkConfig[network.name]['functionsPublicKey']}`)
-    overrides
-      ? await oracle.setDONPublicKey('0x' + networkConfig[network.name]['functionsPublicKey'], overrides)
-      : await oracle.setDONPublicKey('0x' + networkConfig[network.name]['functionsPublicKey'])
-    console.log('DON public key set')
-  
-    console.log('Authorizing oracle with registry')
-    overrides
-      ? await registry.setAuthorizedSenders([oracle.address], overrides)
-      : await registry.setAuthorizedSenders([oracle.address])
-    console.log('Oracle authorized with registry')
-  
-    console.log(`Setting oracle registry to ${registry.address}`)
-    const setRegistryTx = overrides
-      ? await oracle.setRegistry(registry.address, overrides)
-      : await oracle.setRegistry(registry.address)
-    console.log('Oracle registry set')
-  
-    if (process.env.ETHERSCAN_API_KEY || process.env.POLYGONSCAN_API_KEY) {
-      console.log('Waiting 6 blocks before verifying contracts...')
-      await setRegistryTx.wait(6)
+  }
+
+  console.log('Deploying Functions oracle factory')
```

Comment:
Oracle needs to deploy before registry because of allow list change for Automation

Reply:
Good catch!